opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
> a comment to that effect would not go amiss.

okay. I'll add one tomorrow.

> I was more concerned about the correctness implications of a smart pointer parameter.

I thought that with `shared_ptr` everything is golden as long as there are no cycles. If there are, they can still be used, but weak references need to be inserted to break them. maybe I'm wrong, though...